share/shwap/range_namespace_data_id.go

```diff
@@ -0,0 +1,141 @@
+package shwap
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"io"
+)
+
+// RangeNamespaceDataIDSize defines the size of the RangeNamespaceDataIDSize in bytes, combining SampleID size, Namespace size,
+// 2 additional bytes for length of the range and a uint representation of bool flag.
+const RangeNamespaceDataIDSize = SampleIDSize + share.NamespaceSize + 2 + 2
+
+// RangeNamespaceDataID identifies the continuous range of shares in the DataSquare(EDS),
+// starting from the given `SampleID` and contains `Length` number of shares.
+type RangeNamespaceDataID struct {
+	SampleID
+	//  RangeNamespace is a string representation of the namespace of the requested range.
+	RangeNamespace share.Namespace
+	// Length is a number of shares to of the requested range.
+	Length uint16
```

Comment:
In other shwap types we use int in ID struct and convert to the wire type in marshal functions

Reply:
Also, do you think uint16 is enough? Why not uint32?

Reply:
I think uint32 is better. Thanks.